src/main/java/com/fasterxml/jackson/core/util/TextBuffer.java

```diff
@@ -432,36 +432,61 @@ public char[] getTextBuffer()
      *
      * @return Aggregated buffered contents as a {@link java.lang.String}
      */
+    @Deprecated //@since 2.15
     public String contentsAsString()
+    {
+        return contentsAsString(StreamReadConstraints.defaults());
+    }
+
+
+    /**
+     * Accessor that may be used to get the contents of this buffer as a single
+     * {@code String} regardless of whether they were collected in a segmented
+     * fashion or not: this typically require construction of the result String.
+     *
+     * @param constraints constraints for stream reading
+     * @return Aggregated buffered contents as a {@link java.lang.String}
+     * @since 2.15
+     */
+    public String contentsAsString(StreamReadConstraints constraints)
```

Comment:
Ok, I do not think check should be done here: at this point content accumulation has potentially gone way past maximum. Instead it should be checked when requesting new segment in `finishCurrentSegment()`.

This does still tie into quesiton of whether to always pass `SRC` or construct TextBuffer with it -- let's forget that for a moment.

Reply:
I was going to suggest that we take TextBuffer and duplicate the validate methods from StreamReadConstraints as local protected methods and have them be no-op. The base TextBuffer can be used for write cases like SegmentedStringWriter.

We can then add a ConstrainedReadTextBuffer that needs a StreamReadConstraints to construct. The validate methods will be overridden to delegate to StreamReadConstraints.

Reply:
when we create StreamWriteConstraints, we can subclass TextBuffer again.

Reply:
I removed the append checks. finishCurrentSegment is the main check now.